src/main/groovy/io/seqera/wave/controller/ViewController.groovy

```diff
@@ -306,15 +325,55 @@ class ViewController {
 
         // return the response
         binding.put('server_url', serverUrl)
-        return HttpResponse.<Map<String,Object>>ok(binding)
+        return HttpResponse.ok(new ModelAndView("scan-view", binding))
     }
 
+    Map<String,?> renderScansView(List<WaveScanRecord> results) {
+        // create template binding
+        final bindingMap = new HashMap<String, ?>(3)
+        if( results ) {
+            bindingMap.put("scan_container_image", results[0].containerImage)
+            final binding = new ArrayList<Map<String,String>>()
+            for (def result : results) {
+                final bind = new HashMap(20)
+                bind.scan_id = result.id
+                bind.scan_status = result.status
+                bind.scan_time = formatTimestamp(result.startTime) ?: '-'
+                bind.scan_vuls_count = result.status == 'SUCCEEDED' ? result.vulnerabilities.size() : '-'
+                binding.add(bind)
+            }
+            bindingMap.put('scan_records', binding)
+        }
+        // result the main object
+        bindingMap.put('server_url', serverUrl)
+        return bindingMap
+    }
+
+    protected String isScanInvalidSuffix(String scanId) {
+        // check for scan id containing a -nn suffix
+        final check1 = DASH_SUFFIX_REGEX.matcher(scanId)
+        if( check1.matches() ) {
+            return "/view/scans/${check1.group(1)}_${check1.group(2)}"
+        }
+        return null
+    }
+
+    static final private Pattern SCAN_ID_REGEX = ~/((sc-)?[0-9a-z\-]{16})(?<!_\d{2})$/
```

Comment:
```suggestion
    static final private Pattern SCAN_ID_REGEX = ~/((sc-)?[0-9a-z\-]{14,16})(?<!_\d{2})$/
```

It turns out, it can be shorter then 16